ts/main.ts

```diff
@@ -0,0 +1,49 @@
+interface FormElements extends HTMLFormControlsCollection {
+  title: HTMLInputElement;
+  photo: HTMLInputElement;
+  notes: HTMLInputElement;
+}
+
+interface FormObject {
+  entryId?: number;
+  title?: string;
+  photo?: string;
+  notes?: string;
+}
+
+const $idImagUrl = document.querySelector('#photo-url') as HTMLInputElement;
+const $image: any = document.querySelector('.image') as HTMLImageElement;
+const $idJournalEntry = document.querySelector(
+  '#journal-entry',
+) as HTMLFormElement;
+
+$idImagUrl?.addEventListener('input', () => {
+  if (!$idImagUrl) {
+    $image.src = 'images/placeholder-image-square.jpg';
+  }
+  $image.src = $idImagUrl.value;
+});
+
+let nextEntryId: number;
+$idJournalEntry.addEventListener('submit', (event: Event) => {
+  event.preventDefault();
+  const data = readData();
+  const $formElements = $idJournalEntry.elements as FormElements;
+  nextEntryId = data.nextEntryId;
+  const formObject: FormObject = {};
+  formObject.entryId = data.nextEntryId;
+  formObject.title = $formElements.title.value;
+  formObject.photo = $formElements.photo.value;
+  formObject.notes = $formElements.notes.value;
+  data.entries.unshift({
+    entryId: formObject.entryId,
+    title: formObject.title,
+    photo: formObject.photo,
+    notes: formObject.notes,
+  });
```

Comment:
Why recreate the object? Just utilize your variable
```suggestion
  data.entries.unshift(formObject);
```

Reply:
done